src/github.com/getlantern/flashlight/logging/useragents.go

```diff
@@ -0,0 +1,50 @@
+package logging
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"sync"
+)
+
+type agentsMap map[string]int
+
+var (
+	userAgents  = make(agentsMap)
+	agentsMutex = &sync.Mutex{}
+	reg         = regexp.MustCompile("^Go.*package http$")
+)
+
+// registerUserAgent tries to find the User-Agent in the HTTP request
+// and keep track of the applications using Lantern during this session
+func RegisterUserAgent(agent string) {
+	// Do this asynchronously because it is not a critical operation,
+	// so there is no wait for the mutex in the caller goroutine
+	go func() {
+		if agent != "" {
+			agentsMutex.Lock()
+			defer agentsMutex.Unlock()
+			if n, ok := userAgents[agent]; ok {
+				userAgents[agent] = n + 1
+			} else {
+				userAgents[agent] = 1
+			}
+		}
+	}()
+}
+
+// getSessionUserAgents returns the
+func GetSessionUserAgents() string {
```

Comment:
Sure! The reason I did it public is because of its symmetry with the other method, but is not going to be used outside logging in principle.